src/install/index.js

```diff
@@ -191,22 +191,26 @@ const preCleanup = () => {
     });
 };
 
-const postCleanup = (deployDoc) => {
-  return clearStagedDdocs()
-    .then(() => {
-      debug('Delete deploy ddoc');
-      deployDoc._deleted = true;
-      return DB.app.put(deployDoc);
-    })
-    .then(() => {
-      debug('Cleanup old views');
-      return DB.app.viewCleanup();
-    });
+const postCleanup = (deploy, changedApps, deployDoc) => {
+  return Promise.all([
+    deploy.removeOldVersion(changedApps),
+    clearStagedDdocs()
```

Comment:
You can remove old versions and clear staged ddocs at the same time if you want, but deleting the deploy doc is the last thing we should do. Deleting that indicates that horti has finished with the deploy and everything has panned out.

So:
```js
return Promise.all([
  removeOldVersions,
  clearStagedDdocs
])
.then(deleteDeployDocs)
.then(cleanupViews);
```

Or whatever. You could even consider flipping cleaning up old views to beforehand, that's not that big of a deal though, it's just a couchdb compaction style process that has no effect on the running system apart from freeing space.